File: cr.py
import argparse

level_cr_checkpoints = {
    1: 520,
    30: 1220,
    80: 2120,
    150: 3250,
    240: 5025,
    241: 5025
}

recommended_crs = []

level_checkpoints = list(level_cr_checkpoints.keys())
for lower_bound_level in level_checkpoints:
    if level_checkpoints.index(lower_bound_level) < len(level_checkpoints) - 1:
        higher_bound_level = level_checkpoints[level_checkpoints.index(lower_bound_level) + 1]
        levels = higher_bound_level - lower_bound_level

        lower_bound_cr = level_cr_checkpoints[lower_bound_level]
        upper_bound_cr = level_cr_checkpoints[higher_bound_level]
        cr_range = upper_bound_cr - lower_bound_cr
        cr_per_level = cr_range/levels

        for level in range(lower_bound_level, higher_bound_level):
            this_cr = lower_bound_cr + round((level - lower_bound_level) * cr_per_level)
            recommended_crs.append(this_cr)

def get_hell(level):
    if level < 30:
        return 'Hell I'
    elif level < 80:
        return 'Hell II'
    elif level < 150:
        return 'Hell III'
    elif level < 240:
        return 'Hell IV'
    else:
        return 'Hell V'

def get_cr(level):
    return recommended_crs[level - 1]

File: cr.js
function get_hell(level) {
  if (level < 30) {
    return 'Hell I';
  } else if (level < 80) {
    return 'Hell II';
  } else if (level < 150) {
    return 'Hell III';
  } else if (level < 240) {
    return 'Hell IV';
  } else {
    return 'Hell V';
  }
}

function get_cr(level) {
  if (level > 240) {
    return '5025++';
  }

  const levelCrCheckpoints = new Map([
    [1, 520],
    [30, 1220],
    [80, 2120],
    [150, 3250],
    [240, 5025]
  ]);

  const levelCheckpoints = Array.from(levelCrCheckpoints.keys());

  var recommendedCrs = new Map();

  levelCheckpoints.forEach(lowerBoundLevel => {
    var index = levelCheckpoints.indexOf(lowerBoundLevel)
    if (index < (levelCheckpoints.length - 1)) {
      var upperBoundLevel = levelCheckpoints[index + 1];
      var levels = upperBoundLevel - lowerBoundLevel;
      var lowerBoundCr = levelCrCheckpoints.get(lowerBoundLevel);
      var upperBoundCr = levelCrCheckpoints.get(upperBoundLevel);
      var crRange = upperBoundCr - lowerBoundCr;
      var crPerLevel = crRange/levels;

      var levelRange = [...Array(levels).keys()].map(i => i + lowerBoundLevel);
      for (const level of levelRange) {
        var levelDiff = level - lowerBoundLevel;
        var levelCr = lowerBoundCr + Math.ceil(levelDiff * crPerLevel);
        recommendedCrs.set(level, levelCr);
      }
    } else {
      recommendedCrs.set(lowerBoundLevel, levelCrCheckpoints.get(lowerBoundLevel));
    }
  })

  return recommendedCrs.get(level);
}

function display() {
  var level = parseInt(document.paragon.level.value);
  document.getElementById('paragon_level').innerHTML = "<b>Paragon Level:</b> " + level;
  document.getElementById('hell').innerHTML = "<b>Hell Level:</b> " + get_hell(level);
  document.getElementById('cr').innerHTML = "<b>Combat Rating:</b> " + get_cr(level);
}
